### PR TITLE
Support state reason for issues

### LINKED ISSUE #1792: State reason for issues not supported
**Describe the bug**
GitHub supports closing an issue with `state_reason`, but this is not supported in this library at the moment. This is not really a bug, but the issue templates for this repository only allows creating bugs.

**To Reproduce**

**Expected behavior**

**Desktop (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by stianst**
Hi, first of thanks for providing this library. I'm using it quite a lot, but finding a few things that are missing for my needs, and trying to contribute those, with the first one being supporting closing issues as not planned.

Trying to figure out how to work with the testsuite, but having a few issues. 

When running `mvn install -Dtest.github.org=false -Dtest=GHIssueTest` I'm getting the following exception:
```
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field protected java.lang.String java.net.HttpURLConnection.method accessible: module java.base does not "opens java.net" to unnamed module @551bca18
```

Another thing is the test is hardwired to `hub4j-test-org` which I don't have access to, so guess I need to use my own repository somehow.

Closes #1792

### CODE REVIEW

**Comment by gsmet** [comment]
Hey @stianst ,

Yeah, you usually need access to the test org when recording testing as it's easier for them to be reproduced by others when something needs a change.

You need @bitwiseman for that.

As for the error you have when running the tests, the test runs by default with an HTTP Client that is not compatible with Java 17+. Run them with Java 11 and things should work better.

File: src/main/java/org/kohsuke/github/GHIssue.java
Comment (gsmet) [inline src/main/java/org/kohsuke/github/GHIssue.java:72]: This needs an update

File: src/main/java/org/kohsuke/github/GHIssueStateReason.java
Comment (gsmet) [inline src/main/java/org/kohsuke/github/GHIssueStateReason.java:4]: This is off. But I'm not sure if we are encouraged to put our name here, these days. @bitwiseman would know.

File: src/main/java/org/kohsuke/github/GHIssueStateReason.java
Comment (gsmet) [inline src/main/java/org/kohsuke/github/GHIssueStateReason.java:29]: As previously stated, you can add `UNKNOWN` here. This avoids errors when GitHub add a new value.

**Comment by gsmet** [comment]
Ah, actually, good news, I forgot I had the permission to do that.

I added you to the test org. You should have received an invite.

@bitwiseman Stian is a colleague from Red Hat leading the Keycloak project.

**Review by gsmet — CHANGES_REQUESTED** [review]
I spotted a few minor issues.

**Comment by gsmet** [comment]
Ah also, do not record things for all the tests, doing:

```
rm -rf src/test/resources/org/kohsuke/github/GHIssueTest/wiremock/closeIssue/
mvn install -Dtest.github.takeSnapshot -Dtest=GHIssueTest#closeIssue
rm -rf src/test/resources/org/kohsuke/github/GHIssueTest/wiremock/closeIssueNotPlanned/
mvn install -Dtest.github.takeSnapshot -Dtest=GHIssueTest#closeIssueNotPlanned
```

will be better.

Then you should add the new snapshots to your commit.

File: src/main/java/org/kohsuke/github/GHIssueStateReason.java
Comment (stianst) [reply src/main/java/org/kohsuke/github/GHIssueStateReason.java]: Looked at a couple recent commits and they don't have any license headers so removed this.

File: src/main/java/org/kohsuke/github/GHIssue.java
Comment (stianst) [reply src/main/java/org/kohsuke/github/GHIssue.java]: Nice, that's much better. Updated to use that.

**Comment by stianst** [comment]
Got the test working eventually (well, okay I only spent 30 min, so this wasn't too bad ;)), but used an org I created for the purpose. Will try with the official org and update the data.

**Comment by stianst** [comment]
Should be ready to go now

File: src/main/java/org/kohsuke/github/GHIssue.java
Comment (gsmet) [inline src/main/java/org/kohsuke/github/GHIssue.java:213]: Are you sure this can't be null? My understanding is that when the issue is not closed, it will be null but I might be mistaken.
If so, please use `EnumUtils.getNullableEnumOrDefault()`. I pointed you to it in my previous comment but it might have been a bit confusing as I pointed to this one first.

**Review by gsmet — COMMENTED** [review]
I added a very small comment for something that we need to check but otherwise it looks very good.

Note that we will have to wait for @bitwiseman to merge it as I'm not a committer here.

**Comment by bitwiseman** [comment]
> @gsmet @bitwiseman would it be okay to enable fine-grained personal access tokens for the org (https://github.com/organizations/hub4j-test-org/settings/personal-access-tokens-onboarding). They are a bit nicer than the legacy tokens as you can limit to a specific org/repo.

Enabled the fine-grained PATs.

File: src/main/java/org/kohsuke/github/GHIssue.java
Comment (stianst) [reply src/main/java/org/kohsuke/github/GHIssue.java]: Updated, I was kinda thinking it should be set to UNKNOWN when null, but obviously null makes more sense ;)

**Comment by stianst** [comment]
Fixed the last issue with using `EnumUtils.getNullableEnumOrDefault` and added Javadocs to `GHIssueStateReason` since the `site` job complained about that.

**Review by gsmet — APPROVED** [review]

File: src/main/java/org/kohsuke/github/GHIssueStateReason.java
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHIssueStateReason.java:3]:
```suggestion

/**
 * The enum GHIssueStateReason.
 */
public enum GHIssueStateReason {
```

File: src/main/java/org/kohsuke/github/GHIssue.java
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHIssue.java:322]:
```suggestion
    /**
     * Closes this issue.
     *
     * @param reason
     *            the reason the issue was closed
     * @throws IOException
     *             the io exception
     */
    public void close(GHIssueStateReason reason) throws IOException {
```

File: src/main/java/org/kohsuke/github/GHIssueStateReason.java
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHIssueStateReason.java:3]:
```suggestion
```

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**
